agent/consul/gateways/bind.go

```diff
@@ -45,8 +45,8 @@ func BindRoutesToGateways(gateways []*structs.BoundAPIGatewayConfigEntry, routes
 					delete(parentRefs, ref)
 				}
 			} else {
-				if gateway.UnbindRoute(route) {
-					modified = append(modified, gateway)
+				if gateway.unbindRoute(route) {
```

Comment:
can we move this `if` to an `else if` to cut down on the nesting here since there's no other conditions here?